EmailLogCleanup/src/main/java/org/vm/email/cleanup/Clean.java

```diff
@@ -0,0 +1,297 @@
+        int numSendgrid = statement.executeUpdate();
+    
+        update = "DELETE FROM mail_log WHERE guid IN (?)";
+        update = update.replace("(?)", sqlIN);
+        statement = con.prepareStatement(update);
+        int numMailLog = statement.executeUpdate();
+    
+        //Closes connections and statements.
+        stat.execute("SET FOREIGN_KEY_CHECKS=1");
+        statement.close();
+        stat.close();
+        return numSendgrid + numMailLog;
+    }
+    
+    /**
+     * Deletes all emails of type "custom_email_to_volunteer" from the database before specified date.
+     * @param con is the current connection.
+     * @param dateX is the date determining which files are deleted.
+     * @return The total number of rows deleted from this operation.
+     * @throws SQLException in case of error.
+     */
+    public int deleteCustomToVolunteer(Connection con, Timestamp dateX) throws SQLException {
+        Statement stat = con.createStatement();
+        stat.execute("SET FOREIGN_KEY_CHECKS=0");
+        
+        String update = "DELETE mail_log, sendgrid_event FROM mail_log INNER JOIN sendgrid_event " +
+                            "ON mail_log.guid = sendgrid_event.guid WHERE " +
+                            "mail_log.type = 'custom_email_to_volunteer' AND sent_time < ?";
+        PreparedStatement statement = con.prepareStatement(update);
+        statement.setTimestamp(1, dateX);
+        int addNumDeleted = statement.executeUpdate();
+    
+        stat.execute("SET FOREIGN_KEY_CHECKS=1");
+        statement.close();
+        stat.close();
+        return addNumDeleted;
+    }
+    
+    /**
+     * Takes a number in the form of a string representing a number of days and returns
+     * the date that occured given number of days before the current date in the form
+     * of a timestamp.
+     * @param offset as a String that represents a given number of days.
+     * @return Timestamp of the target date.
+     *
+     */
+    private Timestamp convertToTimestamp(String offset) throws ParseException {
+        int offDays = Integer.parseInt(offset);
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.DATE, -offDays);
+        java.util.Date date = cal.getTime();
+        SimpleDateFormat format1 = new SimpleDateFormat("yyyy-MM-dd");
+        String stringDate = format1.format(date);
+        java.util.Date dateJ = format1.parse(stringDate);
+        java.sql.Date sDate = new java.sql.Date(dateJ.getTime());
+        return new Timestamp(sDate.getTime());
+    }
+    
+    /**
+     * A helper method that turns a timestamp into a string for the CSV file.
+     * @param ts Timestamp of email.
+     * @return timestamp in a string format that mimics the database timestamp format.
+     */
+    private String tsToString(Timestamp ts) {
+        Date date = new Date();
+        date.setTime(ts.getTime());
+        return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(date);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.